### ISSUE TITLE
Any update on v2 ETA?

### ISSUE THREAD

**Issue by youshy** [chromedp/chromedp#537]
When can we expect a new version of chromedp? I've seen a ton of questions being responded with `wait for v2`, but do you have any ETA when that might happen? Any way we can help to make it faster?

**Comment by mvdan** [comment]
Sorry, can you give some examples of what issues you mean? There is some work planned, but there is not going to be a v2 per se.

We did a big chunk of work and API overhaul about eight months ago, and work has since been more continuous. We'd also like to rework the API some more again, but there is no ETA there.

If there is a bug in place, or a feature that's missing, patches are generally welcome. But as with API redesigns, there is no ETA for when we'll get to do the work ourselves.

chromedp is likely to remain a v0 until we settle on an API that we're happy with.

**Comment by youshy** [comment]
On top of my head - #180 #251 

The question is more of a `should I prepare my apps for a complete API change`, sorry if it came out a bit harsh.

**Comment by mvdan** [comment]
Oh, apologies if those issues were closed in a confusing way. That feature has been implemented for a while. Just see https://godoc.org/github.com/chromedp/chromedp#ListenTarget, which has two examples. That can be used to listen to any target events, including network ones.

There's also ListenBrowser for browser events.

I don't think there will be another massive API overhaul, at least not as big as the last one. We switched to a context-based API, and that's likely to remain that way. `Executor` and `Run`, and particularly their options, are bit weird though. That's probably what's the most likely to not remain exactly as it is now.
